resources/assets/coffee/react/beatmapset-page/main.coffee

```diff
@@ -25,25 +25,39 @@ BeatmapsetPage.Main = React.createClass
     optionsHash = BeatmapsetPageHash.parse location.hash
     @initialPage = optionsHash.page
 
-    beatmaps = _.keyBy @props.beatmapset.beatmaps.data, (o) -> o.id
+    beatmaps = _.concat @props.beatmapset.beatmaps.data, @props.beatmapset.converts.data
+    beatmaps = _.sortBy beatmaps, ['convert', 'difficulty_rating']
 
-    currentBeatmapId =
-      if optionsHash.beatmapId? && beatmaps[optionsHash.beatmapId]?
-        optionsHash.beatmapId
+    # group beatmaps by playmode and then by beatmap id
+    beatmaps = _.groupBy beatmaps, 'mode'
+    # contains the beatmap ids in their appropriate order
+    beatmapList = {}
```

Comment:
but still, `beatmaps[mode]` is an object, and so it doesn't necessarily respect the sorting done earlier